skbio/util/tests/test_decorator.py

```diff
@@ -72,5 +75,146 @@ def foo(cls):
+                y : int, optional
+
+            """
+            return x + y
+        return f
+
+    def test_function_output(self):
+        f = self._get_f('0.1.0')
+        self.assertEqual(f(1), 43)
+
+    def test_function_docstring(self):
+        f = self._get_f('0.1.0')
+        self.assertTrue('State: Stable as of 0.1.0.' in f.__doc__)
+        f = self._get_f('0.1.1')
+        self.assertTrue('State: Stable as of 0.1.1.' in f.__doc__)
+
+    def test_function_signature(self):
+        f = self._get_f('0.1.0')
+        expected = inspect.ArgSpec(
+            args=['x', 'y'], varargs=None, keywords=None, defaults=(42,))
+        self.assertEqual(inspect.getargspec(f), expected)
+        self.assertEqual(f.__name__, 'f')
+
+    def test_missing_kwarg(self):
+        self.assertRaises(ValueError, stable)
+        self.assertRaises(ValueError, stable, '0.1.0')
+
+
+class TestExperimental(unittest.TestCase):
+
+    def _get_f(self, as_of):
+        @experimental(as_of=as_of)
+        def f(x, y=42):
+            """ Add 42, or something else, to x.
+
+                Parameters
+                ----------
+                x : int, x
+                y : int, optional
+
+            """
+            return x + y
+        return f
+
+    def test_function_output(self):
+        f = self._get_f('0.1.0')
+        self.assertEqual(f(1), 43)
+
+    def test_function_docstring(self):
+        f = self._get_f('0.1.0')
+        self.assertTrue(
+            'State: Experimental as of 0.1.0.' in f.__doc__)
+        f = self._get_f('0.1.1')
+        self.assertTrue(
+            'State: Experimental as of 0.1.1.' in f.__doc__)
+
+    def test_function_signature(self):
+        f = self._get_f('0.1.0')
+        expected = inspect.ArgSpec(
+            args=['x', 'y'], varargs=None, keywords=None, defaults=(42,))
+        self.assertEqual(inspect.getargspec(f), expected)
+        self.assertEqual(f.__name__, 'f')
+
+    def test_missing_kwarg(self):
+        self.assertRaises(ValueError, experimental)
+        self.assertRaises(ValueError, experimental, '0.1.0')
+
+
+class TestDeprecated(unittest.TestCase):
+
+    def _get_f(self, as_of, until, reason):
+        @deprecated(as_of=as_of, until=until, reason=reason)
+        def f(x, y=42):
+            """ Add 42, or something else, to x.
+
+                Parameters
+                ----------
+                x : int, x
+                y : int, optional
+
+            """
+            return x + y
+        return f
+
+    def test_function_output(self):
+        f = self._get_f('0.1.0', until='0.1.4',
+                        reason='You should now use skbio.g().')
+        self.assertEqual(f(1), 43)
+
+    def test_deprecation_warning(self):
+        f = self._get_f('0.1.0', until='0.1.4',
+                        reason='You should now use skbio.g().')
+        # adapted from SO example here: http://stackoverflow.com/a/3892301
+        with warnings.catch_warnings(record=True) as w:
```

Comment:
Nevermind, that won't allow you to test the deprecation message, just that the correct type of warning is raised.